src/core_modules/capture-core/components/WidgetReferral/useReferral.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+import { useEffect, useMemo, useState } from 'react';
+import { useDataQuery } from '@dhis2/app-runtime';
+import { referralStatus } from './constants';
+
+export const useReferral = (programStageId: string) => {
+    const [currentReferralStatus, setReferralStatus] = useState();
+    const [selectedRelationshipTypes, setSelectedRelationshipTypes] = useState([]);
+
+    const { data: programStageData } = useDataQuery(useMemo(() => ({
+        programStageReferral: {
+            resource: 'programStages',
+            id: programStageId,
+            params: {
+                fields:
+                ['referral'],
+            },
+        },
+    }), [programStageId]));
+
+    const { data: relationshipTypeData } = useDataQuery({
+        relationshipTypes: {
+            resource: 'relationshipTypes',
+            params: {
+                fields: 'id,access[*],referral,fromConstraint[*],toConstraint[*]',
+            },
+        },
+    });
```

Comment:
Can we grab the relationship metadata from the indexedDB cache instead? The useDataQuery will currently re-download the data every time you open the new event page. The new event and edit event page are not self-contained at all so I think it's fine if you just use the indexedDB cache from here.